wiki/React-Best-Practices-And-Improvements.md

```diff
@@ -1,78 +1,144 @@
+### Props
 
-# Component File Composition
+Each component should define a props interface immediately above the component definition. 
 
-- It would be great if we configured eslint for automatic import sorting. There are many plugins that would make our job easier - [link](https://levelup.gitconnected.com/how-to-sort-imports-in-react-project-550f5ce70cbf)
-- Component name should be the same as the file name
-- Each component should have interface for props defined which is right above the component definition. Props should be destructured right away.
-- React hooks should be on top of the component and should follow the [hooks rules](https://legacy.reactjs.org/docs/hooks-rules.html)
-- We definitely overuse `// eslint-disable-next-line react-hooks/exhaustive-deps` . Passing exact dependencies as eslint suggest really make sense in 99% of situations. If your dependency array causes an infinite loop, it means that either the logic is poorly designed, or the dependency needs to be wrapped in useMemo/useCallback .
-- We should favour using `useState` or `useEffect` instead of `React.useState` or `React.useEffect`. The same goes for other methods imported from the React package
-- We should keep the component as small and as readable as possible. In case the component requires more complex logic, it's a good idea to move this to util functions in another file.
-- Ideally, we should extract business logic to custom hook, that will be responsible for fetching, storing and manipulating data. This way, we follow single responsibility rule.
-- We should not have two components in single file. It is better to create new component in new file.
-- in `return` we should keep as little logic as possible as this supposed to be markup/UI of the component. Thus, all calculations should be made in the component body, assigned to const and used in return.
-- Likewise when it comes to function handlers. We shouldn't write logic where we should keep markup. Let's define handler functions in component body and use them with onClick, onChange, etc.
+This interface should take the naming structure `ComponentProps`, e.g. `AvatarUploadProps`.
 
-**Example of simple component, child component and custom hook.**
+Props should be immediately destructured within the Component's definition.
+
+### Hooks
+
+In general, hooks and utils should be kept close to the component. If a hook or util function begins to be used by multiple pages or components, it should be moved to our `/utils` and `/helpers` directories.
+
+Ideally, we should extract business logic to custom hooks that are responsible for fetching, storing and manipulating data. This way, we follow the [single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle).
+
+React hooks should be housed either top of the component and should follow the [React hooks rules](https://legacy.reactjs.org/docs/hooks-rules.html).
```

Comment:
to be more precise, invoking react hooks should be on top of the component. Definition of hook should be in separate file